src/WithdrawalQueueManager.sol

```diff
@@ -0,0 +1,353 @@
+        if (availableAmount > pendingRequestedRedemptionAmount) {
+            return availableAmount - pendingRequestedRedemptionAmount;
+        } 
+        
+        return 0;
+    }
+
+    /// @notice Calculates the deficit of redemption assets after accounting for all pending withdrawals.
+    /// @return deficit The amount of deficit redemption assets in the unit of account.
+    function deficitRedemptionAssets() public view returns (uint256) {
+        uint256 availableAmount = redemptionAssetsVault.availableRedemptionAssets();
+        if (pendingRequestedRedemptionAmount > availableAmount) {
+            return pendingRequestedRedemptionAmount - availableAmount;
+        }
+        
+        return 0;
+    }
+
+    /// @notice Withdraws surplus redemption assets to a specified address.
+    function withdrawSurplusRedemptionAssets(uint256 amount) external onlyRole(REDEMPTION_ASSET_WITHDRAWER_ROLE) {
+        uint256 surplus = surplusRedemptionAssets();
+        if (amount > surplus) {
+            revert AmountExceedsSurplus(amount, surplus);
+        }
+        redemptionAssetsVault.withdrawRedemptionAssets(amount);
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  FINALITY  ----------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Checks if a withdrawal request with a given index is finalized.
+     * @param index The index of the withdrawal request.
+     * @return True if the request is finalized, false otherwise.
+     */
+    function withdrawalRequestIsFinalized(uint256 index) public view returns (bool) {
+        WithdrawalRequest memory request = withdrawalRequests[index];
+        return isFinalized(request);
+    }
+
+    /**
+     * @notice Checks if a withdrawal request is finalized.
+     *  A request is finalized if secondsToFinalization time has passed since 
+     *  the withdrawal request was created.        
+     * @param request The withdrawal request to check.
+     * @return True if the request is finalized, false otherwise.
+     */
+    function isFinalized(WithdrawalRequest memory request) public view returns (bool) {
+        return block.timestamp >= request.creationTimestamp + secondsToFinalization;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  VIEWS  -------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Returns the details of a withdrawal request.
+     * @param tokenId The token ID of the withdrawal request.
+     * @return request The withdrawal request details.
+     */
+    function withdrawalRequest(uint256 tokenId) public view returns (WithdrawalRequest memory request) {
```

Comment:
look at the above comment, but it might make sense to just revert here if no withdrawal request with this id exists, instead of returning an empty `WithdrawalRequest`